docs/contributing/code-contributions.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Code Contributions
+---
+
+The beauty of contributing to open source is that you can clone your favorite framework, get it running locally, and test out experiments and changes in real time! Way to feel like a wizard.
+
+This page includes instructions specific to the Gatsby core codebase, blog, and website. To start setting up the Gatsby repo on your machine, check out the page on [setting up your local dev environment](/contributing/setting-up-your-local-dev-environment/). For additional instructions on contributing to the docs, visit the [docs contributions page](/contributing/docs-contributions/).
+
+## Making changes to the website
+
+If you want to make more changes to the website functionality or documentation, that is, change layout components or templates, add sections/pages, follow the steps for [setting up your local dev environment](/contributing/setting-up-your-local-dev-environment/). You can then spin up your own instance of the Gatsby website and make/preview your changes before raising
```

Comment:
You don't need to make same setup (including `gatsby-dev-cli`, etc) to contribute to the website code. I know this was moved from current contributing section, but might be good opportunity to fix our mistake?

First 3 steps from below (contributing to the blog) is enough to get going:

> - Clone [the Gatsby repo](https://github.com/gatsbyjs/gatsby/) and navigate to `/www`
> - Run `yarn` to install all of the website's dependencies.
> - Run `npm run develop` to preview the blog at `http://localhost:8000/blog`.

And people can start hacking on website code.